DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/CourseDelegates/SelectedCourseDetails.cs

```diff
@@ -2,18 +2,50 @@
 {
     using System.Collections.Generic;
     using System.Linq;
-    using DigitalLearningSolutions.Data.Models.Courses;
+    using DigitalLearningSolutions.Data.Models.CourseDelegates;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
 
-    public class SelectedCourseDetails
+    public class SelectedCourseDetails : BaseSearchablePageViewModel
     {
-        public SelectedCourseDetails(int customisationId, IEnumerable<Course> courses, IEnumerable<SearchableCourseDelegateViewModel> delegates)
+        public SelectedCourseDetails(
+            CourseDelegatesData courseDelegatesData,
+            string sortBy,
+            string sortDirection,
+            string? filterBy,
+            int page,
+            Dictionary<string, string> routeData
+        ) : base(null, page, true, sortBy, sortDirection, filterBy, routeData: routeData)
         {
-            Active = courses.Single(c => c.CustomisationId == customisationId).Active;
-            Delegates = delegates;
+            Active = courseDelegatesData.Courses.Single(c => c.CustomisationId == courseDelegatesData.CustomisationId)
+                .Active;
+
+            var sortedItems = GenericSortingHelper.SortAllItems(
+                courseDelegatesData.Delegates.AsQueryable(),
+                sortBy,
+                sortDirection
+            );
+            var filteredItems = FilteringHelper.FilterItems(sortedItems.AsQueryable(), filterBy).ToList();
+            MatchingSearchResults = filteredItems.Count;
+            SetTotalPages();
+            var paginatedItems = GetItemsOnCurrentPage(filteredItems);
+            Delegates = paginatedItems.Select(d => new SearchableCourseDelegateViewModel(d));
+            Filters = CourseDelegateViewModelFilterOptions.GetAllCourseDelegatesFilterViewModels();
         }
-        
-        public bool Active { get; set; }
 
+        public bool Active { get; set; }
         public IEnumerable<SearchableCourseDelegateViewModel> Delegates { get; set; }
+
+        public override IEnumerable<(string, string)> SortOptions { get; } = new[]

```

Comment:
Implemented it here, but it has made the CourseDelegateSortByOption defintion much bulkier. Let me know if we think it would be better as it was before, or perhaps in its own file.